vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -197,7 +198,9 @@ void ReactInstanceWin::Initialize() noexcept {
   // InitUIManager uses m_legacyReactInstance
   InitUIManager();
 
-  Microsoft::ReactNative::DevMenuManager::InitDevMenu(m_reactContext);
+  Microsoft::ReactNative::DevMenuManager::InitDevMenu(m_reactContext, [weakReactHost = m_weakReactHost]() noexcept {
+    winrt::make<Microsoft::ReactNative::ConfigureBundlerDlg>(weakReactHost);
```

Comment:
Consider exposing dialog creation as a static function returning void instead of object creation. Right now it's not apparent when looking at this that the ctor will create UI that controls the lifetime of the instance.